tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,251 @@ fn generate_synthetic(
+        } else {
+            writeln!(writer, "{hex_genesis_config}")?;
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use iroha_genesis::{GenesisSignature, GenesisTransaction};
+
+    use super::*;
+    use crate::crypto::AlgorithmArg;
+
+    const GENESIS_JSON_PATH: &str = "../../configs/swarm/genesis.json";
+
+    fn genesis_signing_works() -> Result<bool> {
+        let keypair_config = crypto::Args {
+            algorithm: AlgorithmArg::default(),
+            private_key: None,
+            seed: None,
+            json: true,
+            compact: false,
+        };
+
+        let mut keypair_json = BufWriter::new(Vec::new());
+        keypair_config.run(&mut keypair_json)?;
+        let keypair: KeyPair = serde_json::from_slice(keypair_json.buffer())?;
+
+        let tmp_keypair_json_file = tempfile::NamedTempFile::new()?;
+        fs::write(tmp_keypair_json_file.path(), keypair_json.buffer())?;
+
+        let chain_id = ChainId::from("0123456");
+        let crypto_genesis_config = SignArgs {
+            algorithm: AlgorithmArg::default(),
+            private_key_string: None,
+            private_key_file: None,
+            public_key_string: None,
+            public_key_file: None,
+            keypair_file: Some(PathBuf::from(tmp_keypair_json_file.path())),
+            chain_id: chain_id.clone(),
+            genesis_file: PathBuf::from_str(GENESIS_JSON_PATH)?,
+            out_file: None,
+            scale: false,
+            hex: true,
+        };
+
+        let mut genesis_buf_writer = BufWriter::new(Vec::new());
+
+        crypto_genesis_config.run(&mut genesis_buf_writer)?;
+
+        let mut encoded_genesis_signature = genesis_buf_writer.buffer().to_owned();
+        encoded_genesis_signature.pop();
+
+        let decoded_signed_genesis_config =
+            GenesisSignature::from_hex_string(&encoded_genesis_signature)?;
+
+        let raw_genesis = RawGenesisBlock::from_path(GENESIS_JSON_PATH)?;
+        let signed_genesis_manually =
+            GenesisTransaction::new_unified(raw_genesis.clone(), &chain_id, &keypair);
+        let signed_genesis_from_config =
+            &GenesisNetwork::try_parse(raw_genesis, decoded_signed_genesis_config)?
+                .into_transaction();
+
+        let cmp = |a: &SignedTransaction, b: &SignedTransaction| {
+            a.metadata() == b.metadata()
+                && a.authority() == b.authority()
+                && a.chain_id() == b.chain_id()
+                && a.time_to_live() == b.time_to_live()
+                && a.instructions() == b.instructions()
+                && a.nonce() == b.nonce()
+                && a.signatures() == b.signatures()
+        };
+        Ok(cmp(
+            &signed_genesis_from_config.0,
+            &signed_genesis_manually.0,
+        ))
+    }
+
+    #[test]
+    fn test_genesis_signing_works() {
+        let result = genesis_signing_works();
+        assert!(result.is_ok_and(|result| result));
+    }
+}
```

Comment:
Please avoid using this pattern with nesting. Just make `#[test]` of `genesis_signing_works` directly (with minor internal changes).

Reply:
Forgot to remove when introduced TempFile into the test